INSTALL.rst

```diff
@@ -84,17 +84,42 @@ Installing from source
 
 If you are interested in contributing to Matplotlib development,
 running the latest source code, or just like to build everything
-yourself, it is not difficult to build Matplotlib from source.  Grab
-the latest *tar.gz* release file from `the PyPI files page
-<https://pypi.org/project/matplotlib/>`_, or if you want to
-develop Matplotlib or just need the latest bugfixed version, grab the
-latest git version, and see :ref:`install-from-git`.
+yourself, it is not difficult to build Matplotlib from source.
 
-Matplotlib can be installed from the source directory with a simple ::
+The easiest way to get the latest development version to start contributing
+is to go to the git `repository <https://github.com/matplotlib/matplotlib>`_
+and run: ::
 
-  python -m pip install .
+  git clone https://github.com/matplotlib/matplotlib.git
 
-We provide a setup.cfg_ file which you can use to customize the build
+If you're developing, it's better to do it in editable mode. The reason why
+is that pytest's test discovery only works for Matplotlib
+if installation is done this way. Also, editable mode allows your code changes
+to be instantly propagated to your library code without reinstalling (though
+you will have to restart your python process / kernel): ::
+
+  pip install -e .
+
+If you're not developing, it can be installed from the source directory with
+a simple ::
+
+  pip install .
+
+To run the tests you will need to install some additional dependencies: ::
+
+  pip install -r requirements/dev/dev.txt
```

Comment:
I'd prefer to have this in `requirements/dev-requirements.txt` (and also pull `requirements/doc/doc-requirements.txt` one level up in a separte PR). There is no need for a separate subdirectory for a single file. Additionally `dev-requirements.txt` is more obviously a requirements file than `dev.txt`.

Side remark: There are no general conventions on naming additional requirements files. We just have to choose something reasonable ourselves.

Reply:
This was my suggestion (I'm bad at naming things).  I like `dev-requirements.txt` better that just `dev.txt`, will also push a commit with this change.